packages/big-design/src/components/ButtonGroup/ButtonGroup.tsx

```diff
@@ -0,0 +1,153 @@
+import { MoreHorizIcon } from '@bigcommerce/big-design-icons';
+import React, { createRef, HTMLAttributes, memo, useCallback, useEffect, useMemo, useState } from 'react';
+
+import { useWindowResizeListener } from '../../hooks';
+import { MarginProps } from '../../mixins';
+import { ButtonProps } from '../Button';
+import { Dropdown } from '../Dropdown';
+import { Flex } from '../Flex';
+
+import { StyledButton, StyledFlexItem } from './styled';
+
+export interface ButtonGroupAction extends Omit<ButtonProps, 'children' | 'iconOnly' | 'iconRight' | 'iconLeft'> {
+  text: string;
+  icon?: React.ReactElement;
+}
```

Comment:
Initially, `Button` has `'iconOnly' | 'iconRight' | 'iconLeft'` properties but, according to the guidelines I can show the icon only when action will be hidden under ellipsis. So, I decide to exclude `'iconOnly' | 'iconRight' | 'iconLeft'` and just add the `icon` property.